Automattic/wp-calypso
# Completion of the next step of the plugin details page in 3 simplest flows

#### Changes proposed in this Pull Request
Implement plugin details page buttons
![image](https://user-images.githubusercontent.com/3422709/118125974-4b906800-b415-11eb-942e-66093b7055fb.png)

Completed the next step of the plugin details page for the flows 1, 8, 9, 10 in flow variations shown below
![image](https://user-images.githubusercontent.com/3422709/118590336-dd0a2c00-b7bf-11eb-9a4e-e99c19cde0eb.png)


#### Testing instructions
Flow 1
* Prepare an account on a business plan with a custom domain set as the primary domain
* Go to the yoast plugins page with the feature flag enabled `/plugins/wordpress-seo/{siteUrl}?flags=marketplace-yoast`
    - eg: http://calypso.localhost:3000/plugins/wordpress-seo/awesome.com?flags=marketplace-yoast
* Click on  Buy Yoast Premium, you should be directed to checkout

Flow 8
* Prepare an account on a business plan **without** a custom domain purchased 
* Go to the yoast plugins page with the feature flag enabled `/plugins/wordpress-seo/{siteUrl}?flags=marketplace-yoast`
    - eg: http://calypso.localhost:3000/plugins/wordpress-seo/awesome.com?flags=marketplace-yoast
* Click on  Buy _Yoast Premium_ 
* you should be directed to the domains page with Yoast premium added to the basket

Flow 9,10
* Same as above but
* Click on  buy _Yoast free_ 
* you should be directed to the domains page with nothing added to the basket
